plugins/statFunctions/countFunctions.php

```diff
@@ -0,0 +1,113 @@
+        if(!is_null($checkSurveyId)) {
+            return $checkSurveyId;
+        }
+        $column = self::_getColumnByQCode($surveyId,$qCode);
+        if(is_null($column)) {
+            if(Permission::model()->hasSurveyPermission($surveyId,'surveycontent')) { // update ???
+                return sprintf(gT("Invalid question code %s"),CHtml::encode($qCode));
+            }
+            return "";
+        }
+        $sQuotedColumn=Yii::app()->db->quoteColumnName($column);
+        $oCriteria = new CDbCriteria;
+        $oCriteria->condition= "$sQuotedColumn IS NOT NULL";
+        $oCriteria->addCondition("submitdate IS NOT NULL");
+        $oCriteria->compare($sQuotedColumn,$comparaison);
+        tracevar([
+            $qCode,
+            intval(SurveyDynamic::model($surveyId)->count($oCriteria))
+        ]);
+        return intval(SurveyDynamic::model($surveyId)->count($oCriteria));
+    }
+
+    /**
+     * Return the count of reponse on current Expression Manager survey equal to a specific value
+     * Can use < or > … see https://www.yiiframework.com/doc/api/1.1/CDbCriteria#compare-detail
+     * @todo
+     * @param string $qCode
+     * @param boolean $submitted response
+     * @return integer|string
+     */
+    public static function statCount($qCode)
+    {
+        $surveyId = LimeExpressionManager::getLEMsurveyId();
+        $checkSurveyId = self::_checkSurveyId($surveyId);
+        if(!is_null($checkSurveyId)) {
+            return $checkSurveyId;
+        }
+        $column = self::_getColumnByQCode($surveyId,$qCode);
+        if(is_null($column)) {
+            return "";
+        }
+
+        $sQuotedColumn=Yii::app()->db->quoteColumnName($column);
+        $oCriteria = new CDbCriteria;
+        $oCriteria->condition= "$sQuotedColumn IS NOT NULL and $sQuotedColumn <> ''";
+        $oCriteria->addCondition("submitdate IS NOT NULL");
+        tracevar([$column,$surveyId,SurveyDynamic::model($surveyId)->count($oCriteria)]);
+        return intval(SurveyDynamic::model($surveyId)->count($oCriteria));
+    }
+
+    /**
+     * Check the survey
+     * @param $surveyId
+     * @return integer|string|null null man surveyId can be used
+     */
+    private static function _checkSurveyId($surveyId)
+    {
+        $oSurvey = Survey::model()->findByPk($surveyId);
+        if(!$oSurvey) {
+            return "Invalid survey"; // Can not happen … (hope)
+        }
+        if(!$oSurvey->getIsActive()) {
+            return 0;
+        }
+    }
+
+    /**
+     * Check the survey
+     * @param $surveyId
+     * @return null|string null mean an invalid column/code
+     */
+    private static function _getColumnByQCode($surveyId,$qCode)
+    {
+        $availableColumns = SurveyDynamic::model($surveyId)->getAttributes();
+        /* Sample : Q01.sgqa Q01_SQ01.sgqa */
+        if(array_key_exists($qCode,$availableColumns)) {
+            return $qCode;
+        }
+        /* @todo : allow "Q0" and "Q0_SQ0" …
```

Comment:
Some are really easy (single choice), some just less simple butbstill easy (Q1_SQ001), some more complex (Q01_other ? Q01comment) etc … 

See https://gitlab.com/SondagesPro/coreAndTools/getQuestionInformation/blob/master/helpers/surveyCodeHelper.php#L74 for sample code to get all …